### ISSUE TITLE
Ability to edit failed templates

### ISSUE THREAD

**Issue by phorcys420** [coder/coder#15834]
After talking about #15035 w/ Ben today we realized that we should improve how we handle failing examples templates.

> This also means that the template won't be added to your instance due to the error, and you can't just change that, so you need to clone and edit the example from the repo, and then push it to your Coder instance.

I think we should make it so that even if the template fails, it will be added to your instance and you will be shown an "Edit" button.
That way, even though the template is still broken, the fixing part will be made easier and users won't be locked to only stare at the error.

cc: @matifali

**Comment by matifali** [comment]
I agree that we should improve this flow in the context of our efforts in #15447.
A similar issue is discussed in #6860.

**Comment by phorcys420** [comment]
In comparison to #15447, I think we should still run terraform plan but create the template with empty resources in a failed state.

Maybe we should grey it out or display a warning sign if people try to create a workspace from it.

**Comment by bpmct** [comment]
@hugodutka another one that may impact user->first agent flow
